pkg/reconciler/pipeline/controller.go

```diff
@@ -144,9 +144,17 @@ func (r *pipelineReconciler) reconcile(ctx context.Context, pl *dfv1.Pipeline) (
 	}()
 
 	pl.Status.SetObservedGeneration(pl.Generation)
-
-	if oldPhase := pl.Status.Phase; pl.Spec.Lifecycle.GetDesiredPhase() == dfv1.PipelinePhasePaused ||
-		oldPhase == dfv1.PipelinePhasePaused || oldPhase == dfv1.PipelinePhasePausing {
+	// Regular pipeline change
+	// This should be happening in call cases to ensure a clean initialization regardless of the lifecycle phase
```

Comment:
```suggestion
	// This should be happening in all cases to ensure a clean initialization regardless of the lifecycle phase
```

Reply:
I assume that's what you meant?

Reply:
Merging this, will address it in another PR.